execution/executor/src/components/block_tree/mod.rs

```diff
@@ -250,8 +261,26 @@ impl BlockTree {
             );
             last_committed_block
         };
-        *self.root.lock() = root;
-        Ok(())
+        let old_root = {
+            let mut root_locked = self.root.lock();
+            // send old root to async task to drop it
+            let old_root = root_locked.clone();
+            *root_locked = root;
+            old_root
+        };
+        // This should be the last reference to old root, spawning a drop to a different thread
+        // guarantees that the drop will not happen in the current thread
+        let (tx, rx) = channel::<()>();
+        rayon::spawn(move || {
+            let _timeer = APTOS_EXECUTOR_OTHER_TIMERS_SECONDS
+                .with_label_values(&["drop_old_root"])
+                .start_timer();
+            drop(old_root);
+            // Error is ignored, since the caller might not care about dropping completion and
+            // has discarded the receiver already.
+            tx.send(()).ok();
```

Comment:
you just want the join handle right?

Reply:
`spawn()` doesn't return one.

Reply:
Spawning a new thread will be slower in theory?

Reply:
I guess it's probably faster than the channel lol

Reply:
The cost of spawning a thread should be ~10us, not too bad.

Reply:
just allocating the channel is not a fair comparison, you need to send something in a separate rayon task lol

Reply:
Yeah but that's not important because it's async and noboday receive in production.

Reply:
Ideally we have a global drop pool

Reply:
it's just ugly but anyway spawning thread is much more expensive than i expected

Reply:
landing ugly for now. maybe creating the drop pool "some time in the future"